### DISCUSSION TITLE
Wird eine zentrale oder eine dezentrale 'Registry' vorgesehen?

### DISCUSSION THREAD

**Discussion by KampfCaspar** [e-id-admin/governance-sounding-board#9] [General]
Leider habe ich die Repos erst jetzt entdeckt - deshalb bitte ich um Entschuldigung, falls der Punkt bereits diskutiert wurde:

In den Architekturskizzen wird eine Registry (wenn auch 'pragmatisch') mit APIs eingezeichnet. Wie 'zentral' soll denn diese Registry aufgebaut sein und welche Funktionen erfüllt sie? In den Extrema ist denkbar:

1. Es handelt sich um einen konkreten zentralen Dienst, der alle Vertrauensanker beinhaltet.
    Die Aufnahme eines Vertrauensankers in den zentralen Dienst ist eng kontrolliert und an Bedingungen geknüpft,
    Issuer, die nicht zentral registriert sind, existieren nicht.
2. Es handelt sich nur um ein dezentrales Konzept - jeder Verifier führt in Tat und Wahrheit eine eigene Registry mit den 
    Ankern der Issuer, die er tatsächlich benötigt.
    Die Architektur der 'Registry' definiert nur, wie die Vertrauensanker zum Verifier kommen - die eigentliche Verteilung der
    Vertrauensanker kann über verschiedene Wege erfolgen.  Beliebige Issuer können ihren Vertrauensanker zum 'Import' anbieten.

An einem einfachen Beispiel erläutert:

X GmbH möchte eigene eMitarbeiterausweise für einen Sicherheitsbereich ausstellen und dafür das eID-Ökosystem nutzen. In Variante 2 kann sie ohne weiteres elektronische Ausweise zur Verfügung stellen, welche die Mitarbeiter z.B. in ihr eID-Wallte aufnehmen. Der Sicherheitsdienst der X GmbH nimmt den Vertrauensanker X GmbH direkt bei sich auf und kann damit innerhalb des bestehenden Ökosystems die eMitarbeiterausweise prüfen. Ausserhalb der X GmbH bleibt der Ausweis 'wertlos'. In Variante 1 müsste die X GmbH sich formell registrieren, bevor sie das Ökosystem nutzen kann.

Der Entscheid läuft wohl darauf hinaus, ob man eine Kontrolle behalten möchte, _wer_ das Ökosystem nutzt - und allenfalls versuchen,  Missbrauch zu verhindern. Umgekehrt wäre eine dezentrale Variante sehr viel datenschutzfreundlicher. IMHO lässt sich mit einer offenen Verwendung, allenfalls mit klar verständlichem Vorgehen zur Frage welche Anker installiert werden, am meisten Nutzen kreieren.

**Comment by grt-dti** [comment]
Die Frage kennen wir doch aus den "alten" PKI-Strukturen - defakto muss das doch am Ende der Verifier entscheiden.

Eine "closed group of issuer" ist doch nichts anderes als eine Einschränkung in der Möglichkeit Vertrauen zu besitzen - in einem Ökosystem wird sich der (liberale) Staat dazu nie durchringen können, weil er dadurch den Wettbewerb staatlich regulieren würde. Sein Eingriff wird sich demzufolge auf den staatlich durchgesetzten Entzug von Vertrauen - Ausschluss aus dem Vertrauenssystem - zurückziehen.
Wir werden also für alles Vertrauen in der Öffentlichkeit entweder nur einen zentralen Dienst (das muss nicht zwingend eine zentrale Infrastruktur sein!) für die -weitgehend freie- Registratur von vertrauenswürdigen Teilnehmern am Ökosystem (sowas wie ein elektronischen Handelsregister) oder eine Hierarchie (Pluralität) von diesen regionalisierten Diensten unter einer Akkreditierung bei einer zentralen Stelle bekommen.
Dazu werden wir eine Vielzahl absolut "freifliegender" Vertrauensanker für die individuelle Aussprache von Vertrauen in einer "closed community" bekommen. Jeder Mensch kann ja selbst entscheiden wem er sonst noch vertrauen möchte.....

Und wer sich mit PKI's auskennt, wird genau dieses dort wiederfinden - weil das alles rein gar nichts mit Technologie zu tun hat, sondern mit der Art&Weise der Aussprache, Beschreibung und Persistierung von Vertrauen.

**Reply by ChristophGraf-SWITCH** [reply]
Hilfreich für die Herstellung des Vertrauens ist aus meiner Sicht die Vertrauensakteure aus der "analogen" Welt in die SSI-Welt zu übertragen. Dass das Handelsregister, das BFS mit der UID, oder Sektorplayer mit Mitgliedschaften ihre bestehende Vertrauensausweise direkt als Issuer in der SSI-Welt die genau gleiche Rolle übernehmen recht zielführend.
In der PKI-Welt wurde dies nicht so bewerkstelligt. Dort sind die "Root-Key-Besitzer" zu zwingend zu nutzenden Trusted Third Parties (mit oder ohne Trust...) geworden und die "gewachsenen" Vertrauensakteure konnten ihre angestammte Rolle nicht mehr direkt wahrnehmen.

**Reply by grt-dti** [reply]
korrekt - das müsste man erreichen. Leider bringen sich gerade die PKI-Anbieter in die Position die "Vertrauenstransporteure" in die SSI-Welt werden / sein zu wollen. Darum müssen wir dafür "streiten", dass die originalen Registrare die Verbindung der physischen mit der SSI-Welt sind.

**Reply by KampfCaspar** [reply]
Die TTP-Problematik von TLS ist nicht auf dieses Szenario übertragbar - bzw. es hat hier nichts zu suchen, wenn das jemand reindrücken will. Es geht eben gerade nicht darum, dass beliebige Endpunkte beliebige Issuer verifizieren, sonden der Verifier will Angaben eines ganz bestimmten Issuers verifizieren. Er wählt den Issuer also im Voraus aus, der Issuer gibt 'sein Zertifikat' selbst raus. Es braucht nur einen geordneten Prozess, wie ich an das 'Zertifikat' des Issuers komme. Dabei kann der Issuer gleichzeitig eine Update-Policy, 'CRLs' und ähnliches selbst mitteilen.

Es spricht IMHO nichts dagegen, den Trust Anchor eines Issuers über einen leicht bedienbaren Prozess zu installieren - wenn ich als Verifier ihn denn eben brauche. Natürlich kann in einer Referenz-App, der Trust Anchor z.B. des Bundes für den ePass vorzuinstallieren. Aber grundsätzlich müsste auch der frei installierbar sein.

Allenfalls kann man in einem sehr stark eingeschränkten Rahmen auf TTP setzen - z.B. wenn der TrustAnchor des ePasses als TTP für TrustAnchor der kantonalen Passbüros herhält.

Eine zentrale Registry braucht es IMHO aber nicht.

**Comment by mdoujak** [comment]
Vielleicht muss die Antwort "beides" sein. 
Die Frage sollte nämlich nicht sein "wollen wir eine dezentrale oder zentrale Registry". Die Frage sollte sein: "wie kann ein Verifier seine Registries so verwalten, dass es verständlich und bequem ist". 
In einem rein dezentralen Setup würde die Verwaltung der Registries dem Verifier übertragen. Diese Idee der Selbstverwaltung gab es schon mit PKI aber die Realität sind ganz Lange Listen von Root CA Zertifikaten, die in jedem Browser enthalten sind, weil kein Benutzer seine Root CAs selbst verwalten will.
Ideal scheint mir darum eine zentrale Registry für zertifizierte Issuer und die Möglichkeit für Holder und Verifier nach Gutdünken weitere Registries hinzuzufügen. Das maximiert den Nutzen für alle Beteiligen, weil auch lokale Lösungen im Ökosystem Platz haben.

**Reply by christianheimann** [reply]
Was verstehst du unter "Holder/Verifier fügen weitere Registries hinzu"? Wie würde das konkret aussehen?

**Reply by ChristophGraf-SWITCH** [reply]
Attraktiv fände ich, wenn die Registry so aufgesetzt wird, dass sie ohne Klimmzüge für gängige Use Cases benutzt werden kann. Dass also Verifier darin vorfinden, was sie für die Verifikation benötigen. Ich gehe da sehr einig mit @mdoujak, dass die Frage nicht "zentral oder dezentral" ist. Beides kann zielführend sein. Wichtig ist, dass eine Governance besteht, die verlässliche Information zugänglich macht. Das ist zentral oder dezentral, aber auch offen oder geschlossen möglich.

**Reply by KampfCaspar** [reply]
Ich gehe soweit überein, dass allenfalls ein 'Meta-TrustAnchor' als 'Registry'  dienen kann (siehe oben zu TLS). Das muss dann aber eine Registry 'unter vielen'  sein - d.h. jeder Issuer kann 'seinen'  TrustAnchor zur 'Registry' ausbauen.

**Comment by mdoujak** [comment]
Sorry für den langen Delay.
Ich habe etwas Reasearch in den https://trustoverip.org Ansatz gemacht. Dabei geht es mir aber weniger darum, wie das Registry Konzept umgesetzt wird, sondern wie es verwendet werden kann.  Mein aktuelles (limitiertes) Verständnis ist folgendes:

1. Die Identitäten werden irgendwo auf einer Blockchain oder einem DLT gespeichert. Das darf jeder einfach tun.

2. Damit jemand eine Rolle (Issuer, Holder, Verifiert) einnehmen kann, genügt der Eintrag in der Blockchain bereits.


Nun zum Thema "Trust". Will ich das meine Identität "in einem bestimmten Mass" vertrauenswürdig ist, dann kann ich mich in einer Registry eintragen lassen. Dabei gelten folgende Regeln:

3. Jede Registry publiziert die Regeln nach denen sie Issuer und Verifier aufnimmt. D.h. ich kann in der Policy der Registry nachlesen, welche Qualität in die Verifikation von Issuern und Verifiern einfliess. Diese Regeln bestimmen also das "Mass an Vertrauen".

Beispiel: Die Schweiz betreibt eine Registry und stellt die Regeln auf nach denen sie Gemeinden und Behörden, welche als Issuer von VCs agieren wollen, verwaltet. Ein anderes Set von Regeln beschreibt die Aufnahmen von Verifiern (Polizei, Grenzwache, Steuerbehörde). Es kann davon ausgegangen werden, dass die Regeln der "Swiss Registry" ein sehr hohes "Mass an Vertrauen" erzeugen.

4. Als Holder kann ich entscheiden, welchen Registries mein Wallet implizit vertrauen soll.

Beispiel: Habe ich die "Swiss Registry" als vertrauenswürdig in meinem Wallet, dann werden die Verifikationen von Issuern und Verifiern automatisch erfolgreich sein. Ob ich VCs von diesen Issuern entgegennehmen will oder VCs an Verifier präsentieren will, das ist immer noch meine Entscheidung.

5. Es gibt keine Limitation wieviele Registries entstehen können.

Beispiel: Es kann eine Registry für Webshops in der Schweiz geben. Es kannn eine Registry für "vertrauenswürdige" Webshops geben. 

6. Jeder Issuer und Verifier kann sich in beliebigen Registries registrieren lassen, wenn er deren Policies für die Aufnahme erfüllt.

**Reply by christianheimann** [reply]
Danke für diesee Auslegeordnung: Ich teile diese Ansicht und werde am Online-Meeting vom 3. März 2022 die Begrifflichkeiten versuchen zu klären.
